protocol/messenger_communities.go

```diff
@@ -992,6 +991,71 @@ func (m *Messenger) leaveCommunity(communityID types.HexBytes) (*MessengerRespon
 	return response, nil
 }
 
+func (m *Messenger) CheckAndDeletePendingRequestToJoinCommunity(sendResponse bool) (*MessengerResponse, error) {
+	sendSignal := false
+
+	pendingRequestsToJoin, err := m.communitiesManager.PendingRequestsToJoin()
+	if err != nil {
+		m.logger.Error("failed to fetch pending request to join", zap.Error(err))
+		return nil, err
+	}
+
+	if len(pendingRequestsToJoin) == 0 {
+		return nil, nil
+	}
+
+	response := &MessengerResponse{}
+	timeNow := uint64(time.Now().Unix())
+
+	for _, requestToJoin := range pendingRequestsToJoin {
+		requestTimeOutClock, err := communities.AddTimeoutToRequestToJoinClock(requestToJoin.Clock)
+		if err != nil {
+			return nil, err
+		}
+
+		if timeNow >= requestTimeOutClock {
+			err := m.communitiesManager.DeletePendingRequestToJoin(requestToJoin)
+			if err != nil {
+				m.logger.Error("failed to delete pending request to join", zap.String("req-id", requestToJoin.ID.String()), zap.Error(err))
+				return nil, err
+			}
+
+			requestToJoin.Deleted = true
+			response.AddRequestToJoinCommunity(requestToJoin)
+
+			// Delete activity centre notification for community admin
+			notification, err := m.persistence.GetActivityCenterNotificationByID(requestToJoin.ID)
+			if err != nil {
+				m.logger.Error("failed to fetch pending request to join", zap.Error(err))
+				return nil, err
+			}
+
+			if notification != nil && !notification.Deleted && notification.Type == ActivityCenterNotificationTypeCommunityMembershipRequest {
+				err = m.persistence.DeleteActivityCenterNotification(types.FromHex(requestToJoin.ID.String()))
```

Comment:
I agree it's not a best practice to execute multiple DB operations without a transaction. I would love to run all DB operations inside a single transaction for this feature but `protocol/communities/persistence` and `protocol/activity_center_persistence` are in a different package. 

It's creating a circular dependency if we try to execute DB operations in a single transaction. Either way, updating the AC table from the same communities package (under persistence) or moving the AC related DB operation to AC go file with receiving transaction pointer. GO has a low tolerance towards circular dependency and will produce a compile-time error.

We need to refactor/move the code to prevent this and have all DB queries running in a single transaction.

I suggest we need to look into this in a separate issue.

Reply:
Thanks for attempting this improvement @smohamedjavid.

> I suggest we need to look into this in a separate issue.

I agree with you, it's too much for this PR, so let's break down the work into smaller chunks.